main/src/cgeo/geocaching/activity/TokenAuthorizationActivity.java

```diff
@@ -0,0 +1,319 @@
+package cgeo.geocaching.activity;
```

Comment:
As said in the related issue, GK service doesn't provide OAuth authentication.
I've tried to be as generic as possible. So next connector that need similar authentication system could be easily implemented. This code is largely inspired from your OAuth implementation.